source/merge-deep.d.ts

```diff
@@ -0,0 +1,55 @@
+/**
+Merge two types deeply into a new type. Keys of the second type overrides keys of the first type.
+
+Use-cases:
+- where intersection types would result in a prop having `never` – if two different types for the same property are used, intersection types assume it's type is `never`. `MergeDeep` allows for safe intersection of two types and having the second passed in type have precedence, hence resulting in no unwanted `never` prop types.
+- merging complex, multi-level interfaces that share common props across different levels
+
+@example
+```
+import {MergeDeep} from 'type-fest';
+
+type Foo = {
```

Comment:
Might be useful to have a real world scenario/something that shows intuitively how the right-hand side "wins" and why that's probably what we'd want?

Reply:
Is that really necessary? In all JS syntax right-hand "wins" – see `Object.assign(foo, bar)` or the spread operator `{...foo, ...bar}`. I think it's pretty intuitive for JS/TS devs.

If you have an idea how to succintly explain this I'm not against including it in the docs.

Reply:
I think in terms of description what you have now is good. I meant more that the example should have fields like name/address etc rather than a b c

Reply:
Oh, ok. I updated the example with a use case I found on lodash issues.